robot_controllers_interface/src/controller_manager.cpp

```diff
@@ -184,65 +184,118 @@ int ControllerManager::requestStop(const std::string& name)
 void ControllerManager::update(const ros::Time& time, const ros::Duration& dt)
 {
   // Reset handles
-  for (JointHandleList::iterator j = joints_.begin(); j != joints_.end(); j++)
-    (*j)->reset();
+  for (const auto& joint_handle: joints_)
+    joint_handle->reset();
 
   // Update controllers
-  for (ControllerList::iterator c = controllers_.begin(); c != controllers_.end(); c++)
+  for (const auto& c: controllers_)
   {
-    (*c)->update(time, dt);
+    c->update(time, dt);
   }
 }
 
 void ControllerManager::reset()
 {
   // Update controllers
-  for (ControllerList::iterator c = controllers_.begin(); c != controllers_.end(); c++)
+  for (const auto& c: controllers_)
   {
-    (*c)->reset();
+    c->reset();
   }
 }
 
-bool ControllerManager::addJointHandle(JointHandlePtr& j)
+bool ControllerManager::addJointHandle(JointHandlePtr& joint_handle_ptr)
 {
-  // TODO: check for duplicate names?
-  joints_.push_back(j);
+  // If we don't already have a handle with
+  // a duplicate name, we add it
+
+  for (const auto& joint_handle: joints_)
+  {
+    if (joint_handle->getName() == joint_handle_ptr->getName())
```

Comment:
here we assume that joint handles that have already been added are always non-null.   I'd say throw exeception if someone tries adding a null pointer instead of having to check for nulls everywhere

Reply:
std::invalid_argument

Reply:
I just returned `false` instead because this does have a `true` `false` return value